src/azure-cli-core/azure/cli/core/adal_authentication.py

```diff
@@ -246,3 +231,27 @@ def _timestamp(dt):
     # https://docs.python.org/3/library/unittest.mock-examples.html#partial-mocking
     # https://williambert.online/2011/07/how-to-unit-testing-in-django-with-mocking-and-patching/
     return dt.timestamp()
+
+
+def aad_error_handler(error: dict):
+    """ Handle the error from AAD server returned by ADAL or MSAL. """
+    login_message = ("To re-authenticate, please {}. If the problem persists, "
+                     "please contact your tenant administrator."
+                     .format("refresh Azure Portal" if in_cloud_console() else "run `az login`"))
+
+    # https://docs.microsoft.com/en-us/azure/active-directory/develop/reference-aadsts-error-codes
+    # Search for an error code at https://login.microsoftonline.com/error
+    msg = error.get('error_description')
+
+    from azure.cli.core.azclierror import AuthenticationError
+    raise AuthenticationError(msg, login_message)
+
+
+def adal_error_handler(err: adal.AdalError):
+    """ Handle AdalError. """
+    try:
+        aad_error_handler(err.error_response)
+    except AttributeError:
+        # In case of AdalError created as
+        #   AdalError('More than one token matches the criteria. The result is ambiguous.')
+        raise CLIError(str(err))
```

Comment:
Is it possible to use one of the new [Error Types](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md#available-error-types)?

Reply:
This error is caused by a known unknown issue in ADAL (#15320). As we are migrating to MSAL, this ADAL issue is not planned to be investigated and fixed. @houk-ms which exception do you prefer to use?

Reply:
As discussed offline, raising `UnknownError` for now and also providing comment and a `recommendation`:

https://github.com/Azure/azure-cli/blob/c62df4d79ac61fc40a857897a5b6e2d83ed17830/src/azure-cli-core/azure/cli/core/adal_authentication.py#L255-L259